frontend/src/views/omni/InstallationMedia/Steps/TalosVersion.vue

```diff
@@ -5,9 +5,115 @@ Use of this software is governed by the Business Source License
 included in the LICENSE file.
 -->
 <script setup lang="ts">
+import { watchOnce } from '@vueuse/core'
+import { compare } from 'semver'
+import { computed, onBeforeMount } from 'vue'
+
+import { Runtime } from '@/api/common/omni.pb'
+import type { TalosVersionSpec } from '@/api/omni/specs/omni.pb'
+import type { JoinTokenStatusSpec } from '@/api/omni/specs/siderolink.pb'
+import {
+  DefaultNamespace,
+  DefaultTalosVersion,
+  JoinTokenStatusType,
+  TalosVersionType,
+} from '@/api/resources'
+import TSelectList from '@/components/common/SelectList/TSelectList.vue'
+import { useWatch } from '@/components/common/Watch/useWatch'
+import { useDocsLink } from '@/methods'
 import type { FormState } from '@/views/omni/InstallationMedia/InstallationMediaCreate.vue'
 
-defineModel<FormState>({ required: true })
+const formState = defineModel<FormState>({ required: true })
+
+const { data: talosVersionList, loading: talosVersionsLoading } = useWatch<TalosVersionSpec>({
+  runtime: Runtime.Omni,
+  resource: {
+    type: TalosVersionType,
+    namespace: DefaultNamespace,
+  },
+})
+
+const { data: joinTokenList, loading: joinTokensLoading } = useWatch<JoinTokenStatusSpec>({
+  runtime: Runtime.Omni,
+  resource: {
+    type: JoinTokenStatusType,
+    namespace: DefaultNamespace,
+  },
+})
+
+const talosVersions = computed(() =>
+  talosVersionList.value
+    .filter((v) => !v.spec.deprecated)
+    .map((v) => v.spec.version!)
+    .sort(compare),
+)
+
+const joinTokens = computed(() => joinTokenList.value.map((t) => t.metadata.id!))
+
+// Form defaults
+onBeforeMount(() => (formState.value.talosVersion ??= DefaultTalosVersion))
+watchOnce(joinTokens, (v) => (formState.value.joinToken ??= v[0]))
+
+const whatsNewDocsLink = useDocsLink('talos', "/getting-started/what's-new-in-talos", () => ({
+  talosVersion: formState.value.talosVersion,
+}))
+const k8sSupportMatrixDocsLink = useDocsLink('talos', '/getting-started/support-matrix', () => ({
+  talosVersion: formState.value.talosVersion,
+}))
 </script>
 
-<template>Talos Version</template>
+<template>
+  <div class="space-y-4">
+    <TSelectList
+      v-model="formState.talosVersion"
+      :disabled="talosVersionsLoading"
+      :values="talosVersions"
+      title="Choose Talos Linux Version"
+      overhead-title
+    />
+
+    <p class="text-xs">
+      We strongly recommend using the latest stable version of Talos Linux ({{
+        DefaultTalosVersion
+      }}).
+      <br />
+      Pre-release versions are suitable for testing purposes but are not advised for production
```

Comment:
by default Omni never shows pre-release versions, so this warning is a bit useless most of the time

Reply:
Copied it off designs, not sure if there was some other thought process behind it

@Unix4ever wdyt?